client/src/utils/localStorageHelpers.ts

```diff
@@ -0,0 +1,26 @@
+const CONTEXT = 'context';
+
+export type Context = {
+    protoEnabled?: boolean;
+    promethusEnabled?: boolean;
+}
+
+export function getContextItem(item: string) {
+    return getItem(CONTEXT)[item];
+}
+
+export function setContext(context: Context) {
+    setItem(CONTEXT, {
+        protoEnabled: true,
+        promethusEnabled: false,
+        ...context,
+    });
+}
+
+export function setItem(item: string, object: Object) {
+    localStorage.setItem(item, JSON.stringify(object));
+}
+
+export function getItem(item: string) {
+    return JSON.parse(localStorage.getItem(item) || '');
```

Comment:
I think calling JSON.parse on an empty string will error. I think this needs to be `JSON.parse(localStorage.getItem(item) || '{}');`